File: imagen_sdk/enums.py
from enum import Enum


class PhotographyType(Enum):
    """Photography types for AI optimization (from API spec)."""

    NO_TYPE = "NO_TYPE"
    OTHER = "OTHER"
    PORTRAITS = "PORTRAITS"
    WEDDING = "WEDDING"
    REAL_ESTATE = "REAL_ESTATE"
    LANDSCAPE_NATURE = "LANDSCAPE_NATURE"
    EVENTS = "EVENTS"
    FAMILY_NEWBORN = "FAMILY_NEWBORN"
    BOUDOIR = "BOUDOIR"
    SPORTS = "SPORTS"
    SCHOOL = "SCHOOL"


class CropAspectRatio(Enum):
    """Crop aspect ratios (from API spec)."""

    RATIO_2X3 = "2X3"
    RATIO_4X5 = "4X5"
    RATIO_5X7 = "5X7"


class DNGCompression(Enum):
    """Compression mode for HDR-merged DNG output (from API spec)."""

    LOSSY = "LOSSY"
    LOSSLESS = "LOSSLESS"


class ProjectSource(Enum):
    """Project source for AI enhancement / copilot operations (from API spec)."""

    REGULAR = "REGULAR"
    I2I = "I2I"


class ClientType(Enum):
    """Client type used when creating projects / upload links (from API spec)."""

    APP = "APP"
    API = "API"
    MOBILE = "MOBILE"
    WEB_REAL_ESTATE = "WEB_REAL_ESTATE"

File: sdks/go/enums.go
package imagen

// PhotographyType selects AI optimization tuned to a shoot type.
type PhotographyType string

const (
	PhotographyTypeNoType          PhotographyType = "NO_TYPE"
	PhotographyTypeOther           PhotographyType = "OTHER"
	PhotographyTypePortraits       PhotographyType = "PORTRAITS"
	PhotographyTypeWedding         PhotographyType = "WEDDING"
	PhotographyTypeRealEstate      PhotographyType = "REAL_ESTATE"
	PhotographyTypeLandscapeNature PhotographyType = "LANDSCAPE_NATURE"
	PhotographyTypeEvents          PhotographyType = "EVENTS"
	PhotographyTypeFamilyNewborn   PhotographyType = "FAMILY_NEWBORN"
	PhotographyTypeBoudoir         PhotographyType = "BOUDOIR"
	PhotographyTypeSports          PhotographyType = "SPORTS"
	PhotographyTypeSchool          PhotographyType = "SCHOOL"
)

// CropAspectRatio is the target aspect ratio when cropping is enabled.
type CropAspectRatio string

const (
	CropAspectRatio2x3 CropAspectRatio = "2X3"
	CropAspectRatio4x5 CropAspectRatio = "4X5"
	CropAspectRatio5x7 CropAspectRatio = "5X7"
)

// DNGCompression controls compression of HDR-merged DNG output.
type DNGCompression string

const (
	DNGCompressionLossy    DNGCompression = "LOSSY"
	DNGCompressionLossless DNGCompression = "LOSSLESS"
)

// ProjectSource identifies which project family an enhancement/copilot call
// targets. It is required in enhance, copilot, reset and finalize request bodies.
type ProjectSource string

const (
	ProjectSourceRegular ProjectSource = "REGULAR"
	ProjectSourceI2I     ProjectSource = "I2I"
)
